pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -220,20 +220,22 @@ func getChaosRunnerENV(cr *litmuschaosv1alpha1.ChaosEngine, aExList []string, Cl
 // getChaosRunnerLabels return the labels required for chaos-runner
 func getChaosRunnerLabels(cr *litmuschaosv1alpha1.ChaosEngine) map[string]string {
 	return map[string]string{
-		"app": cr.Name,
-        "chaosUID": string(cr.UID),
+		"app":                         cr.Name,
+		"chaosUID":                    string(cr.UID),
 		"app.kubernetes.io/component": "chaos-runner",
-        "app.kubernetes.io/part-of": "litmus",
+		"app.kubernetes.io/part-of":   "litmus",
 	}
 }
 
-
 // newGoRunnerPodForCR defines a new go-based Runner Pod
 func newGoRunnerPodForCR(engine *chaosTypes.EngineInfo) (*corev1.Pod, error) {
+	engine.VolumeOpts.VolumeOperations(engine.Instance.Spec.Components.Runner.ConfigMaps, engine.Instance.Spec.Components.Runner.Secrets)
+
 	containerForRunner := container.NewBuilder().
 		WithEnvsNew(getChaosRunnerENV(engine.Instance, engine.AppExperiments, analytics.ClientUUID)).
 		WithName("chaos-runner").
 		WithImage(engine.Instance.Spec.Components.Runner.Image).
+		WithVolumeMountsNew(engine.VolumeOpts.VolumeMounts).
```

Comment:
As we are returning an error inside the builder function if `VolumeMounts` is nil. Do we have to put this field as optional, allow only if `engine.VolumeOpts.VolumeMounts` is not nil?